mbtools/inject_ib_uuids.py

```diff
@@ -0,0 +1,38 @@
+import argparse
+from pathlib import Path
+from mbtools import utils
+from uuid import uuid4
+from bs4 import BeautifulSoup
+
+
+CLASSES_TO_INJECT = [
+    "os-raise-ib-input",
+    "os-raise-ib-pset",
+    "os-raise-ib-pset-problem"]
+
+
+def inject_ib_uuids(html_dir):
+    for path in Path(html_dir).rglob('*.html'):
+        with open(path, 'r') as f:
+            soup = BeautifulSoup(f.read(), 'html.parser')
+            for tag in CLASSES_TO_INJECT:
+                for elem in soup.find_all(class_=tag):
+                    if elem.get('data-content-id') is None:
+                        elem['data-content-id'] = uuid4()
+
+            utils.write_html_soup(path, soup)
```

Comment:
Minor: We may want to outdent this so it happens outside of the context manager which has the file open for read. Otherwise this code path writes a file that is open for reading here.